### PR TITLE
Cclient API initial functions.

### LINKED ISSUE #55: Implement/Design `init_iota_http_service`

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR fixed #55 
* Implemented `iota_client_core_init` and `iota_client_core_destroy` 
* Fixed `BUILD` dependence 
* Simplified marcos in json test cases. 

Example for using cclient core API:
```
int main(){
  retcode_t ret = RC_OK;
  iota_client_service_t serv;
  serv.http.host = "localhost";
  serv.http.port = 14265;
  serv.http.api_version = 1;
#ifdef _DEBUG_STDOUT_
  logger_init();
  logger_output_register(stdout);
  logger_output_level_set(stdout, LOGGER_DEBUG);
#endif
  iota_client_core_init(&serv);

  // emit getNodeInfo to IRI
  get_node_info_res_t* node_res = get_node_info_res_new();
  ret = iota_api_get_node_info(&serv, &node_res);
  get_node_info_res_free(&node_res);

  iota_client_core_destroy();
  return ret; 
}
```

# Test Plan:
CI must pass

### CODE REVIEW

File: cclient/BUILD [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-    ]),
+    srcs = [
+        "iota_api.c",
+        "iota_api.h",
+    ],
     hdrs = ["iota_api.h"],
     visibility = ["//visibility:public"],
     deps = [
-        ":service",
+        "//cclient:service",
```
Comment (tsvisabo) [inline cclient/BUILD]: why full path?

File: cclient/service.c [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "service.h"
+#include "serializer/json/json_serializer.h"
+#include "utils/logger_helper.h"
+
+void iota_client_service_init(iota_client_service_t* serv) {
+  // enable logger
+  logger_init_json_serializer();
```
Comment (tsvisabo) [inline cclient/service.c]: should be configurable?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/BUILD [hunk 3]
```diff
@@ -1,28 +1,28 @@
 cc_library(
     name = "api",
-    srcs = glob([
-        "*.c",
-        "**/*.c",
-        "**/*.h",
-    ]),
+    srcs = [
+        "iota_api.c",
+        "iota_api.h",
```
Comment (thibault-martinez) [inline cclient/BUILD]: Header in the sources ?

File: cclient/BUILD [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
+        "//cclient:service",
         "//cclient/http:beast",
     ],
 )
 
 cc_library(
     name = "service",
     srcs = [
+        "service.c",
         "service.h",
```
Comment (thibault-martinez) [inline cclient/BUILD]: Header in the sources ?

File: cclient/service.h [hunk 5]
```diff
@@ -20,27 +20,16 @@ extern "C" {
...
-typedef struct {
-  char* data;
-  size_t length;
-} iota_response_t;
+  http_info_t http;
+  serializer_t serializer;
+} iota_client_service_t;
 
+void iota_client_service_init(iota_client_service_t* serv);
+void iota_client_service_destroy();
```
Comment (thibault-martinez) [inline cclient/service.h]: newline

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/BUILD (hunk 1 above)
Comment (oopsmonk) [reply cclient/BUILD]: I thought it's no difference in this case, isn't it?

File: cclient/service.c (hunk 2 above)
Comment (oopsmonk) [reply cclient/service.c]: not for now.

File: cclient/BUILD (hunk 3 above)
Comment (oopsmonk) [reply cclient/BUILD]: https://docs.bazel.build/versions/master/be/c-cpp.html#cc_binary.srcs

File: cclient/BUILD (hunk 4 above)
Comment (oopsmonk) [reply cclient/BUILD]: https://docs.bazel.build/versions/master/be/c-cpp.html#cc_binary.srcs

File: cclient/BUILD (hunk 3 above)
Comment (thibault-martinez) [reply cclient/BUILD]: Yes but you have them in srcs AND in hdrs so it's kind of redundant

File: cclient/BUILD (hunk 1 above)
Comment (tsvisabo) [reply cclient/BUILD]: exactly :) ,  but i think it's better to use `:` so people who read this file knows the rule is present

File: cclient/BUILD (hunk 3 above)
Comment (oopsmonk) [reply cclient/BUILD]: Ah, I got what you mean, thanks.

**Review by tsvisabo — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Comment by th0br0** [comment]
We're probably missing a couple `const`ness modifiers here, but let's merge and revisit that later! Yay, CClient is usable :party_parrot:

**Pull request merged**